File: l3/Piecokat.cpp
#include "Piecokat.hpp"
#include <cmath>

namespace Zadanie {
    Piecokat::Piecokat(float a)
            : mA(a) {
    }

    float Piecokat::area() {
        return static_cast<float>(5 * std::pow(mA, 2) * (std::cos(M_PI / 5.0) / std::sin(M_PI / 5.0)) / 4);
    }

    float Piecokat::length() {
        return 5 * mA;
    }

    bool Piecokat::validate(float a) {
        if (a > 0)
            return true;
        else
            return false;
    }

}

File: l3/z1/Piecokat.java
public class Piecokat extends Figura{
    Piecokat(float a) {
        this.mA = a;
        this.mFigureName = "pięciokąt";
    }

    @Override
    public float area() {
        return (float)(5 * Math.pow(mA, 2) * (Math.cos(Math.PI / 5.0) / Math.sin(Math.PI / 5.0)) / 4);
    }

    @Override
    public float length() {
        return 5 * mA;
    }

    static boolean validate(float a) {
        return a > 0;
    }

    public String getFigureName() {
        return mFigureName;
    }


    private float mA = 0.f;

    private String mFigureName = "";
}
